### PR TITLE
Expose important event

### PR DESCRIPTION
**Pull request by samplise**
### What changes were proposed in this pull request?

Add API to expose important information.

### Why are the changes needed?

Users can understand the execution of training job in detail.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Unit test.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -32,7 +32,7 @@
 
 setup(
     name="dlrover",
-    version="0.3.6rc0",
+    version="0.3.6.1rc0",
```
Comment (workingloong) [inline setup.py:35]: "0.3.6rc1" is better than "0.3.6.1rc0".

File: dlrover/python/master/elastic_training/rdzv_manager.py [hunk 2]
```diff
@@ -281,6 +282,14 @@ def join_rendezvous(
             self._node_rdzv_times[node_rank] = round(
                 self._lastcall_time - self._start_rdzv_ts, 2
             )
+            if self._error_monitor:
+                self._error_monitor.report_event(
+                    "info",
+                    node_id,
+                    "job rendezvous",
```
Comment (BalaBalaYi) [inline dlrover/python/master/elastic_training/rdzv_manager.py:289]: All the key metrics need to be defined as constant. Better use capital letter to define the metrics.

File: dlrover/python/master/monitor/speed_monitor.py [hunk 3]
```diff
@@ -43,7 +44,7 @@ def __init__(self):
 class SpeedMonitor(object):
     """Monitor the training speed by the number of batch per second"""
 
-    def __init__(self):
+    def __init__(self, error_monitor: ErrorMonitor = None):
```
Comment (BalaBalaYi) [inline dlrover/python/master/monitor/speed_monitor.py:47]: Why error_monitor in speed_monitor?  The implementation of this two monitor is parallel i think.

File: dlrover/python/master/node/dist_job_manager.py [hunk 4]
```diff
@@ -241,9 +243,10 @@ def should_early_stop(self):
                 "2) alive worker number consistently less than the min "
                 "training nodes required 3) pending time last exceed limit."
             )
-            self._error_monitor.process_error(
-                None, 0, msg, level=TrainingExceptionLevel.ERROR
-            )
+            if self._error_monitor:
```
Comment (BalaBalaYi) [inline dlrover/python/master/node/dist_job_manager.py:246]: Should abstract this into a function:
```
def report_failure(self, node, restart_cnt, msg, level):
    if self._error_monitor:
        self._error_monitor.process_error(node, restart_cnt, msg, level)
```
To ignore if-else every where.

**Review by BalaBalaYi — CHANGES_REQUESTED** [review]

File: dlrover/python/master/monitor/error_monitor.py [hunk 5]
```diff
@@ -60,6 +71,16 @@ def process_error(
             logger.error(error_data)
         return False
 
+    def report_event(
+        self,
+        event_type: str,
+        instance: str,
+        action: str,
```
Comment (BalaBalaYi) [inline dlrover/python/master/monitor/error_monitor.py:78]: Need to define a metric constant for all the event action.
e.g.
```
RDZV_PREFIX = "rdzv_"
RDZV_EVENT1 = RDZV_PREFIX + "event1"

WORKER_PREFIX = "worker_"
WORKER_CREATION = "WORKER_PREFIX" + "creation"
```

File: setup.py (hunk 1 above)
Comment (BalaBalaYi) [reply setup.py]: just 0.3.7

**Review by BalaBalaYi — APPROVED** [review]
LGTM

**Pull request merged**